server/src/interfaces.ts

```diff
@@ -0,0 +1,172 @@
+/**
+ * Task describes a single action
+that should be done by the user. For example: “Go for a 10
+minutes run” or “Go to bed before 11:00 PM”
+*/
+interface Task {
+  id: number;
+  description: string;
+}
+
+/**
+ * Achievement describes a set of
+several tasks accomplished in
+the specific way. For example:
+“Complete each task 7 days in
+a row” or “Complete 5 tasks
+before 8:00 AM”
+ */
+interface Achievement {
+  id: number;
+  description: string;
+  image: string;
+  checkComplete: boolean;
+}
+
+/**
+ * Status describes a state of
+some item (a task or an
+achievement) and a
+timestamp, when this state
+was updated
+ */
+interface Status {
+  state: 'Pending' | 'Success' | 'Failure';
+  updated: Date;
+}
+
+/**
+ * Challenge describes a 30-days
+period, during which randomly
+chosen 30 tasks and 5
+achievements are assigned for
+the user. Starting from the first
+day, the user will receive a
+new task every day, which
+should be completed before
+the midnight, overwise it will
+be marked as failed.
+Achievements status is
+calculated based on tasks
+completion. After 30 days the
+challenge could be successful
+(>= 90% tasks completed) or
+failed (<90% tasks completed)
+ */
+interface Challenge {
+  id: number;
+  state: 'In Progress' | 'Success' | 'Failure';
+  startDate: Date;
+  tasksOrder: number[];
+  tasksStatus: boolean;
+  achievementsStatus: boolean;
+}
+
+/**
+ * ArchiveItem describes a task
+and its status for all past tasks
+in the challenge
+ */
+interface ArchiveItem extends Task {
+  status: boolean;
+}
+
+/**
+ * ActualTask provides
+information about a task and
+its current status in scope of
+the challenge
+ */
+interface ActualTask extends Task {
+  status: boolean;
+}
+
+/**
+ * ActualAchievement provides
+information about an
+achievement and its current
+status in scope of the
+challenge
+ */
+interface ActualAchievement extends Task {
+  image: string;
+  status: boolean;
+}
+
+/**
+ *
+ * @param id - id of current challenge
+ *
+ * @returns Returns a current task with its status by the challenge id
+ */
+// declare function getCurrentTask(id: number): Task;
+function getCurrentTask(id: number): Task {
```

Comment:
Should move functions to a separate file. But I believe your task here is not to declare the functions but rather interfaces
like this
```
  type GetCurrentTask = (id: number) => Task
```